applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/flow/FlowTests.kt

```diff
@@ -739,117 +744,172 @@ class FlowTests {
 
     @Test
     fun `Notary - Uniqueness client service flow is finishing without exceptions`() {
-        val requestID =
-            startRpcFlow(
-                bobHoldingId,
-                mapOf(),
-                "net.cordapp.testing.testflows.UniquenessCheckTestFlow"
-            )
+        val requestID = startRpcFlow(
+            bobHoldingId,
+            mapOf(),
+            "net.cordapp.testing.testflows.UniquenessCheckTestFlow"
+        )
         val result = awaitRpcFlowFinished(bobHoldingId, requestID)
         assertThat(result.flowStatus).isEqualTo(RPC_FLOW_STATUS_SUCCESS)
     }
 
-    // TODO CORE-7939 For now this flow WILL succeed, however, this will require modifications once the ledger has been
-    //  finalised. Specifically, create a state before trying to spend it.
     @Test
-    fun `Notary - Non-validating plugin executes successfully and returns signatures`() {
-        val requestID =
-            startRpcFlow(
-                bobHoldingId,
-                emptyMap(),
-                "net.cordapp.testing.testflows.NonValidatingNotaryTestFlow"
-            )
-        val result = awaitRpcFlowFinished(bobHoldingId, requestID)
-        assertThat(result.flowStatus).isEqualTo(RPC_FLOW_STATUS_SUCCESS)
-        assertThat(result.flowResult)
-            .isEqualTo("Received 1 signatures from the notary, plugin ran successfully.")
+    fun `Notary - Non-validating plugin executes successfully when using issuance transaction`() {
+        issueStatesAndValidateResult(3) { issuanceResult ->
+            // 1. Make sure the states were issued
+            assertThat(issuanceResult.flowStatus).isEqualTo(RPC_FLOW_STATUS_SUCCESS)
+
+            val flowResultMap = issuanceResult.mapFlowJsonResult()
+
+            @Suppress("unchecked_cast")
+            assertThat((flowResultMap["issuedStateRefs"] as List<String>)).hasSize(3)
```

Comment:
```suggestion
            assertThat((flowResultMap["issuedStateRefs"] as List<*>)).hasSize(3)
```

We're only checking size, so can't we do the same as we're doing for the input / ref state checks below, and remove the unchecked cast suppression? Could also add this to the `assertAll` block.